sqlmesh/core/macros.py

```diff
@@ -1157,6 +1158,151 @@ def deduplicate(
     return query
 
 
+@macro()
+def date_spine(
+    evaluator: MacroEvaluator,
+    datepart: exp.Expression,
+    start_date: exp.Expression,
+    end_date: exp.Expression,
+) -> t.Union[exp.Alias, exp.Select]:
+    """Returns an aliased QUERY to build a date spine with the given datepart, and range of start_date and end_date. Useful for joining as a date lookup table.
+
+    Args:
+        datepart: The datepart to use for the date spine: day, week, month, year
+        start_date: The start date for the date spine in format YYYY-MM-DD
+        end_date: The end date for the date spine in format YYYY-MM-DD
+
+    Example:
+        >>> from sqlglot import parse_one
+        >>> from sqlglot.schema import MappingSchema
+        >>> from sqlmesh.core.macros import MacroEvaluator
+        >>> sql = "@date_spine('week', '2022-01-20', '2024-12-16')"
+        >>> MacroEvaluator(dialect='duckdb').transform(parse_one(sql)).sql()
```

Comment:
Let's also use `dialect="duckdb"` in both `parse_one` and `sql`. Or we could get rid of it in the `MacroEvaluator` instance?

Reply:
I got rid of it